### ISSUE TITLE
Update documentation with issues using react-test-renderer

### ISSUE THREAD

**Issue by mrchief** [jamesmfriedman/rmwc#250]
Creating this issue as per our [discussion](https://gitter.im/react-material-web-components/Lobby?at=5b290c69ce3b0f268d29bacf) on gitter.

react-test-renderer doesn't support refs based components for testing. See:

[facebook/react/issues/9244](https://github.com/facebook/react/issues/9244)
[facebook/jest/issues/1805](https://github.com/facebook/jest/issues/1805)
[facebook/jest/issues/5462](https://github.com/facebook/jest/issues/5462)

The workarounds are:

### Mock 3rd party components

E.g. 

```
jest.mock('rmwc/TopAppBar/TopAppBar', () => {
  return (props) => <div>{props.children}</div>;
})
```

### Use something like `createNodeMock`

See https://reactjs.org/docs/test-renderer.html#ideas ([source](https://github.com/facebook/jest/issues/5462#issuecomment-363533870))

### Use ReactDOM or Enzyme 

See [facebook/jest/issues/5462#issuecomment-363532494](https://github.com/facebook/jest/issues/5462#issuecomment-363532494)

**Comment by mrchief** [comment]
Enzmye based test:

```
describe('<Header />', () => {
  it('renders', () => {
    const props = {
      siteTitle: 'Test Site Title'
    }

    const wrapper = shallow(<Header {...props} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
```

**Comment by jamesmfriedman** [comment]
@mrchief does the shallow rendering from the Enzyme based test work?

**Comment by mrchief** [comment]
Yes, no issues with Enzyme. I'm still using the polyfill though. Let me know if you want me to try without the polyfill.

**Comment by jamesmfriedman** [comment]
Updated the docs accordingly, will be released with 1.8
